src/backend/distributed/commands/foreign_server.c

```diff
@@ -269,7 +269,7 @@ static List *
 GetObjectAddressByServerName(char *serverName, bool missing_ok)
 {
 	ForeignServer *server = GetForeignServerByName(serverName, missing_ok);
-	Oid serverOid = server->serverid;
```

Comment:
We have not get a segfault so far, bacause all callers calls the method with missing_ok set to false.
But, for the PR #6088 we will be calling the method with missing_ok set to true. Therefore, we should be ready for it now.